seatunnel-engine/seatunnel-engine-client/src/main/java/org/apache/seatunnel/engine/client/JobConfigParser.java

```diff
@@ -211,43 +230,57 @@ private void initRelationMap(List<? extends Config> sourceConfigs, List<? extend
      * |
      * Sink
      */
-    private List<Action> sampleAnalyze(List<? extends Config> sourceConfigs,
-                                       List<? extends Config> transformConfigs,
-                                       List<? extends Config> sinkConfigs) {
-        SeaTunnelSource seaTunnelSource = ConnectorInstanceLoader.loadSourceInstance(sourceConfigs.get(0));
+    private void sampleAnalyze(List<? extends Config> sourceConfigs,
+                               List<? extends Config> transformConfigs,
+                               List<? extends Config> sinkConfigs) {
+        ImmutablePair<SeaTunnelSource, List<URL>> pair =
+            ConnectorInstanceLoader.loadSourceInstance(sourceConfigs.get(0));
         SourceAction sourceAction =
-            new SourceAction(idGenerator.getNextId(), seaTunnelSource.getPluginName(), seaTunnelSource);
+            createSourceAction(idGenerator.getNextId(), pair.getLeft().getPluginName(), pair.getLeft(),
+                pair.getRight());
         sourceAction.setParallelism(getSourceParallelism(sourceConfigs.get(0)));
+
+        ImmutablePair<SeaTunnelSink<SeaTunnelRow, Serializable, Serializable, Serializable>, List<URL>>
+            sinkListImmutablePair;
+
         if (transformConfigs.size() != 0) {
```

Comment:
using Collection.isEmpty() makes the code more readable and can be more performant. The time complexity of any isEmpty() method implementation should be O(1) whereas some implementations of size() can be O(n).